doni/driver/worker/ironic.py

```diff
@@ -215,13 +215,14 @@ def process(
         state_details: "dict" = None,
     ) -> "WorkerResult.Base":
         hw_props = hardware.properties
+        hw_capabilities = hw_props.get("baremetal_capabilities")
 
-        capabilities_string = ",".join(
-            [
-                f"{key}:{value}"
-                for key, value in hw_props.get("baremetal_capabilities").items()
-            ]
-        )
+        if isinstance(hw_capabilities, Mapping):
```

Comment:
almost... the `typing` library is used for type hinting, aka things that shouldn't actually affect or get called at runtime. I'm surprised that this actually evaluates to True properly, maybe there's some odd reason for this, or they have coded it in anticipation or in reaction to it being called like this. I think `isinstance(hw_capabilities, dict)` is sufficient, but what if it's empty? Then the capabilities string will be the empty string, is that OK?

Reply:
good catch. I'd meant to reference `collections.abc.Mapping` there. Technically this is coming from a FlexibleDict field, so maybe I don't need to check type here? Wasn't sure if that would always return a `dict` vs an OrderedDict or something else.